tests/core/factory/test_validation_definition_factory.py

```diff
@@ -308,6 +310,60 @@ def _test_validation_definition_factory_delete_success(
         context.validation_definitions.get(name)
 
 
+@pytest.mark.filesystem
+def test_validation_definition_factory_round_trip(
+    empty_data_context: FileDataContext,
+    validation_definition: ValidationDefinition,
+):
+    # Arrange
+    context = empty_data_context
+
+    ds = context.sources.add_pandas("my_ds")
+    csv_path = (
+        pathlib.Path(__file__).parent.parent.parent
+        / "test_sets"
+        / "quickstart"
+        / "yellow_tripdata_sample_2022-01.csv"
+    )
+    assert csv_path.exists()
+    asset = ds.add_csv_asset("my_asset", filepath_or_buffer=csv_path)
+
+    batch_definition = asset.add_batch_definition("my_batch_def")
+    suite = ExpectationSuite(
+        name="my_suite",
+        expectations=[
+            gxe.ExpectColumnValuesToBeBetween(column="passenger_count", min_value=0, max_value=10),
+            gxe.ExpectColumnMeanToBeBetween(
+                column="passenger_count",
+                min_value=0,
+                max_value=1,
+            ),
+        ],
+    )
+
+    # Act
+    validation_definition = ValidationDefinition(
+        name="my_validation_def", data=batch_definition, suite=suite
+    )
+    persisted_validation_definition = context.validation_definitions.add(
+        validation=validation_definition
+    )
+    retrieved_validation_definition = context.validation_definitions.get(
+        name=validation_definition.name
+    )
+
+    # Assert
+    assert (
+        retrieved_validation_definition.suite.to_json_dict()
+        == persisted_validation_definition.suite.to_json_dict()
+    )
```

Comment:
Equality is a little weird with suites so we have to do this and break up different asserts